spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/RowDataReader.java

```diff
@@ -195,5 +195,10 @@ protected StructLike asStructLike(InternalRow row) {
     protected InputFile getInputFile(String location) {
       return RowDataReader.this.getInputFile(location);
     }
+
+    @Override
+    protected void markRowDeleted(InternalRow row) {
+      row.setBoolean(columnIsDeletedPosition(), true);
```

Comment:
Yes. That's the assumption. Agreed that we'd better not have that assumption. There will be a bit more expensive since we need to set every row now. I'd guess the perf impact is minor though. Will make the change.

Reply:
We should test the perf implication but we will remove the if statement so it should be either on par or faster.

Reply:
Discussed with @aokolnychyi offline a bit. I've tested out that we actually set the default value explicitly here, https://github.com/apache/iceberg/blob/664439344aefc60836fd519a21c57c0406c3f96c/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/data/SparkParquetReaders.java#L170. So it is not an assumption. Would prefer to keep it as is.